### ISSUE TITLE
<title> content is being parsed, but should be text only for non-XML docs

### ISSUE THREAD

**Issue by billneff79** [fb55/htmlparser2#482]
The content of the `<title>` tag should be 
> Text that is not inter-element whitespace.

Using version 4.1.0 of htmlparser2, the content is being parsed as though it were tags.  
**Example Document**
```html
<head>
    <title><b>foo</b></title>
</head>
```
**Expected Outcomes**
1. `onopentag` and `onclosetag` are called for the `head` and `title` nodes
1. `ontext` is called with the content of `<b>foo</b>` as the text content of the `title` node

**Actual Outcomes**
1. `onopentag` and `onclosetag` are called for the `head`, `title`, and `b` nodes
1. `ontext` is called with the content of `foo` as the text content of the `b` node

**Comment by ecammit** [comment]
+1

**Comment by billneff79** [comment]
Here is the link to the `title` specification: https://www.w3.org/TR/html52/document-metadata.html#the-title-element

It has not changed since the original DOM Level 1 specification.  To quote some portions  directly here for reference:
```
Content model:
Text that is not inter-element white space.

DOM interface:
interface HTMLTitleElement : HTMLElement {
  attribute DOMString text;
};
```
>

**Comment by fb55** [comment]
Fixed in #483
